packages/ethereum-contracts/contracts/superfluid/SuperToken.sol

```diff
@@ -676,6 +676,28 @@ contract SuperToken is
         _approve(account, spender, amount);
     }
 
+    function operationIncreaseAllowance(
+        address account,
+        address spender,
+        uint256 addedValue
+    )
+        external override
+        onlyHost
+    {
+        _approve(account, spender, _allowances[account][spender] + addedValue);
+    }
+
+    function operationDecreaseAllowance(
+        address account,
+        address spender,
+        uint256 subtractedValue
+    ) 
+        external override
+        onlyHost
+    {
+        _approve(account, spender, _allowances[account][spender] - subtractedValue);
```

Comment:
The allowance might go negative here? That's not allowed by ERC-20.



Please consult: 
* https://github.com/OpenZeppelin/openzeppelin-contracts/blob/f8e3c375d19bd12f54222109dd0801c0e0b60dd2/contracts/token/ERC20/ERC20.sol#L197
* https://github.com/superfluid-finance/protocol-monorepo/blob/e7fe12f0ca6b860b98b338372d9282a8cb599005/packages/ethereum-contracts/contracts/superfluid/SuperToken.sol#L433 (can't we call this instead of `_approve` and effectively re-implementing?)

Reply:
allowance can go negative in OZ's ERC-20 implementation because they have the subtraction in an `unchecked` block. we are using SafeMath and the SuperToken contract does not inherit from OZ's ERC20 implementation

Reply:
I'm still slightly confused why we effectively have 2 implementations for the same thing in the same file:
* https://github.com/superfluid-finance/protocol-monorepo/blob/e7fe12f0ca6b860b98b338372d9282a8cb599005/packages/ethereum-contracts/contracts/superfluid/SuperToken.sol#L433
* https://github.com/superfluid-finance/protocol-monorepo/blob/e7fe12f0ca6b860b98b338372d9282a8cb599005/packages/ethereum-contracts/contracts/superfluid/SuperToken.sol#L690
No way to re-use the logic?

Reply:
So it's expected that we have the 2 slightly different implementations in the same file?
<img width="502" alt="image" src="https://user-images.githubusercontent.com/10894666/224684416-0523f0f9-0446-46e5-8abd-62c2e4e294fc.png">
<img width="506" alt="image" src="https://user-images.githubusercontent.com/10894666/224684474-87d64c31-fed3-4791-9ae9-58409cfb610f.png">

Reply:
nah, initially I removed safecast, but didn't change the new implementation to use it as well